electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+"""
+    single_particle_electrode.py
+
+    Class file for single-particle (SPM) electrode methods
+"""
+
+import cantera as ct
+import numpy as np
+
+def initialize(input_file, inputs, sep_inputs, counter_inputs, electrode_name, 
+        params, offset):
+    """
+    Initialize the model.
+    """ 
+    class electrode:
+        """
+        Create an electrode object representing the dense electrode
+        """
+
+        # Import relevant Cantera objects.
+        bulk_obj = ct.Solution(input_file, inputs['bulk-phase'])
+        elyte_obj = ct.Solution(input_file, inputs['electrolyte-phase'])
+        conductor_obj = ct.Solution(input_file, inputs['conductor-phase'])
+        surf_obj = ct.Interface(input_file, inputs['surf-phase'], 
+            [bulk_obj, elyte_obj, conductor_obj])
+
+        # Anode or cathode? Positive external current delivers positive charge 
+        # to the anode, and removes positive charge from the cathode.
+        name = electrode_name
+        if name=='anode':
+            i_ext_flag = -1
+        elif name=='cathode':
+            i_ext_flag = 1
+        else:
+            raise ValueError("Electrode must be an anode or a cathode.")
+
+        # Store the species index of the Li ion in the Cantera object for the 
+        # electrolyte phase:
+        index_Li = elyte_obj.species_index(inputs['mobile-ion'])
+
+        # Electrode thickness and inverse thickness:
+        dy = inputs['thickness']
+        dyInv = 1/dy
+
+        # Phase volume fractions
+        eps_solid = inputs['eps_solid']
+        eps_elyte = 1 - eps_solid
```

Comment:
You likely don't need this, since `eps_elyte` will vary with time.  You will evaluate it directly at each time step as:

`eps_elyte = 1. - eps_solid - eps_oxide`

where `eps_oxide` varies with time.

Reply:
I think in Amy's model eps_solid changes and there is no eps_oxide, so this what I copied over in main.  Is making an eps_oxide prefered